build/escargot.cmake

```diff
@@ -3,7 +3,7 @@ CMAKE_MINIMUM_REQUIRED (VERSION 2.8)
 # FLAGS
 SET (ESCARGOT_CXXFLAGS)
 SET (ESCARGOT_LDFLAGS)
-SET (ESCARGOT_LIBRARIES)
+SET (ESCARGOT_LIBRARIES checked_arithmetic gc rapidjson)
```

Comment:
Our other project which imports escargot also uses the same gc module. That project builds escargot library and gc library together. so, gc build is included only for escargot executable until now. It seems somewhat complex, but gc need to be separated.

Reply:
Or if you change some newly added features such as ```BUILD_BYPRODUCTS''' to that of old version, I'll fix the other project's build system to fit with this pr later. Thank you.

Reply:
I agree that gc needs to be separated. One of the goals of this PR was to separate it even better than the current approach. That's why it turned `gc` from a custom target into an imported library (built as an external project). Having it as a proper library enables the use of the interface include directory concept of cmake, lifting the need for manually setting include paths by `-I...`, etc.

Additionally, it seems to me that gc is actually a dependency of the escargot library. The linking of the shell binary needs it of course, but that's because the engine calls the functions of the gc lib. If gc is a library (in cmake terms) and the engine lib depends on it, plus the shell depends on the engine lib, then cmake will handle the transitivity and will add the gc lib as a link dependency to the shell binary as well.

That being said, there must surely be something that's not that obvious and causes the conflict.